test/unit/network/ConnectionRegistryTest.js

```diff
@@ -99,6 +164,94 @@ describe('ConnectionRegistryTest', function () {
                 expect(anotherConnection).to.be.equal(firstConnection);
 
                 expect(loadBalancerStub.next.called).to.be.false;
+                expect(loadBalancerStub.nextDataMember.called).to.be.false;
+            }
+        );
+
+        it('should return data member connection when one exists and when data member is requested, [dummy mode]',
+            function () {
+                const firstUUID = UuidUtil.generate();
+                const secondUUID = UuidUtil.generate();
+                const thirdUUID = UuidUtil.generate();
+
+                const loadBalancerStub = {next: sandbox.spy(), nextDataMember: sandbox.spy()};
+                const clusterServiceStub = {};
+                clusterServiceStub.getMember = sandbox.stub();
+
+                clusterServiceStub.getMember.withArgs(firstUUID.toString()).returns({
+                    liteMember: true
+                });
+                clusterServiceStub.getMember.withArgs(secondUUID.toString()).returns({
+                    liteMember: false
+                });
+                clusterServiceStub.getMember.withArgs(thirdUUID.toString()).returns({
+                    liteMember: true
+                });
+
+                const connectionRegistry = new ConnectionRegistryImpl(
+                    new ConnectionStrategyConfigImpl(),
+                    false,
+                    loadBalancerStub,
+                    clusterServiceStub
+                );
+
+                const secondConnection = {};
+                const firstConnection = {};
+                connectionRegistry.setConnection(firstUUID, firstConnection);
+                connectionRegistry.setConnection(secondUUID, secondConnection);
+                connectionRegistry.setConnection(thirdUUID, {});
+
+                const connection = connectionRegistry.getRandomConnection();
+                const otherConnection = connectionRegistry.getRandomConnection();
+                const dataMemberConnection = connectionRegistry.getRandomConnection(true);
+
+                expect(connection).to.be.equal(firstConnection);
```

Comment:
Does this a reference or object equality check? If it an object equality check, then all connections are the same